src/payment/coupon/data/sagas.js

```diff
@@ -39,10 +32,13 @@ export function* handleRemoveCoupon(action) {
   yield put(removeCouponBegin());
   try {
     const result = yield call(deleteCoupon, action.payload.id);
-    yield put(fetchBasketSuccess(result));
+    yield call(handleBasketResult, result);
     yield put(removeCouponSuccess(result));
```

Comment:
Does this remove a coupon success message?  Should that happen before handling the newly returned result?